STYLE: fix pylint redefined-outer-name warnings (#49656)

grtcoder opened the pull request:
Fixed warnings for the following files :-
- **pandas/core/arrays/datetimelike.py**
- **pandas/core/base.py**
- **pandas/core/computation/pytables.py**

- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

grtcoder committed:
STYLE: fix pylint redefined-outer-name warnings (#49656)

Fixed warnings for the following files :-
- pandas/core/arrays/datetimelike.py
- pandas/core/base.py
- pandas/core/computation/pytables.py

MarcoGorelli commented on pandas/core/arrays/datetimelike.py:
```diff
@@ -118,7 +118,7 @@
 from pandas.core.algorithms import (
     checked_add_with_arr,
     isin,
-    mode,
+    mode as mode_func,
     unique1d,
 )
```
in a few other files, there is
```python
import pandas.core.algorithms as algos
```

shall we just do that and then call `algos.mode`, etc...?

grtcoder replied:
Sure, Thanks!

grtcoder replied:
Hey are these two algos the same ?
```python
from pandas._libs import (
    algos,
    lib,
)
```
```python
from pandas.core.algorithms import algos

```

MarcoGorelli replied:
ah good catch - let's just do
```
from pandas.core import algorithms 
```
and then `algorithms.mode`, etc...

grtcoder replied:
Done

MarcoGorelli commented on pandas/core/base.py:
```diff
@@ -62,7 +62,7 @@
 )
 from pandas.core.accessor import DirNamesMixin
 from pandas.core.algorithms import (
-    duplicated,
+    duplicated as duplicated_func,
     unique1d,
     value_counts,
 )
```
same here

grtcoder replied:
Sure, Thanks !

grtcoder replied:
Done

MarcoGorelli commented on pandas/core/computation/pytables.py:
```diff
@@ -22,7 +22,7 @@
 from pandas.core.computation import (
     expr,
-    ops,
+    ops as ops_func,
```
instead of aliasing this one, can we just rename the variable in
```python
ops = PyTablesExprVisitor.binary_ops + PyTablesExprVisitor.unary_ops + ("=",)
```
? `operations` should be fine

grtcoder replied:
Done

MarcoGorelli reviewed (changes requested):
thanks for working on this!

grtcoder commented:
I have amended the commit in the base branch. Is that usually the workflow, or do people prefer separate commits for iterations?

MarcoGorelli commented on pandas/core/arrays/datetimelike.py:
```diff
@@ -1690,7 +1690,7 @@ def _mode(self, dropna: bool = True):
             mask = self.isna()
 
-        i8modes = mode(self.view("i8"), mask=mask)
+        i8modes = algorithms.algos.mode(self.view("i8"), mask=mask)
```
```suggestion
        i8modes = algorithms.mode(self.view("i8"), mask=mask)
```

MarcoGorelli reviewed (changes requested):
adding commit to the same PR is the preferred approach

grtcoder committed:
STYLE: fix pylint redefined-outer-name warnings (#49656)

Fixed warnings for the following files :-
 - pandas/core/arrays/datetimelike.py
 - pandas/core/base.py
 - pandas/core/computation/pytables.py
Iteration :- 2

grtcoder committed:
Merge branch 'pandas-dev:main' into enable_pylint

MarcoGorelli reviewed (approved):
thanks @grtcoder 

![](https://i.fluffy.cc/27MHQpGVHf8zsBqx96gxT5PB5mS1jtjD.gif)
